File: codes/ruby/004.Loops.rb
# while loop

i = 1
while true
    if (i == 2)
        i += 1
        break if (i > 5)
        next
    end
    print("while loop v1, i is #{i}\n")
    i += 1
    break if (i > 5)
end

i = 1
while (i <= 5)
    if (i == 2)
        i += 1
        next
    end
    print("while loop v2, i is #{i}\n")
    i += 1
end

# for loop

# start.f(stop,step)
(1.step(5, 1)).each do |i| # we can also replace `(1.step(5, 1))` with `(1..5)`, but beware that ruby can't do `(5..1)`, use (5.step(1, -1)) instead
    next if (i == 2)
    print("for loop ascending, i is #{i}\n")
end

# start.f(stop,step)
(5.step(1, -1)).each do |i|
    next if (i == 2)
    print("for loop descending, i is #{i}\n")
end

File: codes/julia/004.Loops.jl
using Printf
using Statistics

# while loop

i = 1
while true
    global i
    if (i === 2)
        i += 1
        if (i > 5)
            break
        end
        continue
    end
    println("while loop v1, i is ", i)
    i += 1
    if (i > 5)
        break
    end
end

i = 1
while (i <= 5)
    global i
    if (i === 2)
        i += 1
        continue
    end
    println("while loop v2, i is ", i)
    i += 1
end

# for loop

for i in 1:1:5 # start:step:stop
    if (i === 2)
        continue
    end
    println("for loop ascending, i is ", i)
end

for i in 5:-1:1 # start:step:stop
    if (i === 2)
        continue
    end
    println("for loop descending, i is ", i)
end
